### PR TITLE
fix 'body.appendChild' method works incorrectly in the particular case' (close #421)

### LINKED ISSUE #421: Call to 'body.appendChild' with 'body' from an another iframe inserts element to wrong 'body'
When an iframe is initializing [through IFRAME_DOCUMENT_CREATED_EVENT](https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/sandbox/node/index.js#L97), a top-level `DocumentSandbox` will be attached to it. So, when you call methods like `iframe.contentDocument.open`, top-level sandboxes will be reattached to the `iframe` on [DOCUMENT_CLEANED_EVENT](https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/sandbox/index.js#L141), including `shadowUI`. So if you try to call  `window.top.document.body.appendChild(child)`,  `child` may be inserted into `iframe.contentDocument.body`, because `child` [will be inserted](https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/sandbox/node/element.js#L229) before parent of `ShadowUI.root` which attached to `iframe`.

### PR DESCRIPTION
**Pull request by miherlosev**
/cc @AlexanderMoskovkin @AndreyBelym @LavrovArtem

### CODE REVIEW

File: test/client/fixtures/sandbox/iframe-test.js [hunk 1]
```diff
@@ -332,3 +333,51 @@ test("native methods of the iframe document aren't overridden for iframe with ja
...
+    var createIframes                 = function () {
+        for (var i = 0; i < countIframes; i++) {
+            var iframe = document.createElement('iframe');
+
+            iframes.push(iframe);
+            iframe.id = 'test_GH_i_421_' + i;
+            iframe.setAttribute('src', 'javascript:false');
+            document.body.appendChild(iframe);
+        }
+    };
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/sandbox/iframe-test.js]: add an empty line after each multiline var declaration

**Comment by miherlosev** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Comment by AndreyBelym** [comment]
please, wait until I'll check how it works for the tests on ASPxHTMLEditor demo page. If they will work, then lgtm.

File: test/client/fixtures/sandbox/iframe-test.js [hunk 2]
```diff
@@ -332,3 +333,53 @@ test("native methods of the iframe document aren't overridden for iframe with ja
...
+        for (var i = 0; i < countIframes; i++) {
+            var iframe = iframes[i];
+
+            iframe.contentDocument.open();
+            iframe.contentDocument.write('<html><head><style></style><title>title</title></head><body></body></html>');
+            iframe.contentDocument.close();
+        }
+    };
+
+    var checkIframes                  = function () {
```
Comment (AndreyBelym) [inline test/client/fixtures/sandbox/iframe-test.js]: Oops, misaligned.

File: test/client/fixtures/sandbox/iframe-test.js [hunk 3]
```diff
@@ -332,3 +333,53 @@ test("native methods of the iframe document aren't overridden for iframe with ja
     strictEqual(iframeSandbox._shouldSaveIframeNativeMethods(iframe), browserUtils.isWebKit);
 });
 
+test("'body.appendChild' method works incorrectly in the particular case (GH-421)", function () {
+    var iframes                       = [];
+    var countIframes                  = 2;
+    var createIframes                 = function () {
```
Comment (AndreyBelym) [inline test/client/fixtures/sandbox/iframe-test.js]: This declaration looks ugly IMHO, wouldn't it be better if will be separated by new line with `function` keyword aligned close to equal sign?

**Comment by AndreyBelym** [comment]
Fix works, that's nice, thank you. But I found minor style issues, so \r-, sorry.

File: test/client/fixtures/sandbox/iframe-test.js (hunk 3 above)
Comment (AlexanderMoskovkin) [inline test/client/fixtures/sandbox/iframe-test.js]: agree

**Comment by miherlosev** [comment]
FPR

**Comment by AndreyBelym** [comment]
fine, lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**